### DISCUSSION TITLE
Please stop hiding PR Comments under "Hidden Items" when a PR gets too many comments or commits

### DISCUSSION THREAD

**Discussion by blairio** [community/community#112603] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

Sometimes a PR gets long with lots of commits and lots of comments through the review process. I can understand hiding the commits in the middle of a timeline within a "Hidden Items Load More" button but comments are usually important and need to be reviewed, searched for and followed up.  If they are hidden they can be easily missed and then communication breaks down. Please stop hiding comments automatically within the "Hidden Items... Load more" thing.
Thanks!

**Comment by pastk** [comment]
I second that.

The same problem applies to issues also.
A set of messages might get hidden under "Hidden Items... Load more" and hence they become unsearchable by a usual page search (and I didn't find another way to search inside a particular issue/PR).

**Comment by ChrisMacGregorONI** [comment]
https://github.com/orgs/community/discussions/130618#discussioncomment-12105463

**Comment by DanielGibson** [comment]
See also my reply at https://github.com/orgs/community/discussions/130618#discussioncomment-12195939
which describes why they way "Load more…" is implemented is even more annoying than a feature like this would have to be (you have to click it until *all* comments are shown to see more of the latest comments, and commits, for example from rebase + forcepush, make it show even fewer recent comments)

**Comment by medmunds** [comment]
Since this doesn't seem likely to change any time soon, here is an alert box you can add to PRs where it has become a problem:

> <img width="360" height="216" alt="This PR has hidden comments! GitHub has hidden some comments on this page, which may include unresolved review feedback and other open discussions. Be sure to search the page for “Load more…” and click all the links found." src="https://github.com/user-attachments/assets/9096e077-1b73-447d-b5be-4c8d48cfa795" />
>
> Upvote [*Please stop hiding PR comments*](https://github.com/orgs/community/discussions/112603) to request a fix.

Copy and paste this markdown into the PR:

```markdown
<img width="360" height="216" alt="This PR has hidden comments! GitHub has hidden some comments on this page, which may include unresolved review feedback and other open discussions. Be sure to search the page for “Load more…” and click all the links found." src="https://github.com/user-attachments/assets/9096e077-1b73-447d-b5be-4c8d48cfa795" />

Upvote [*Please stop hiding PR comments*](https://github.com/orgs/community/discussions/112603) to request a fix.
```

**Reply by cb88** [reply]
This wouldn't be so annoying if there was both an unhide hide all button link so for 400 comment PRs you could just read them all its VERY common for PRs with lots of comments to have dozens of comments a day and getting to them is currently a massive pain because all you'll see is a few comments at the end and not all of the last dozen comments.

**Comment by brandon-fryslie** [comment]
It's a performance issue for GitHub.  They aren't going to change it.  I'd recommend having Claude or someone write a chrome extension to click the links for you.

**Comment by DanielGibson** [comment]
They could at least add an additional button/link to load more posts starting at the end, because right now to see the latest hidden comments you have to click "load more" until **all** comments are shown - this increases the load for Github.
So if this poor multibillion dollar company really needs to cripple their usability for performance issues, it's in  their own interest to make this improvement.
